shader/flatten.wgsl

```diff
@@ -188,96 +357,118 @@ fn flatten_cubic(cubic: CubicPoints, path_ix: u32, local_to_device: Transform, o
                 length(vec2(mat.x - mat.w, mat.y + mat.z));
     }
 
-    let err_v = 3.0 * (p2 - p1) + p0 - p3;
-    let err = dot(err_v, err_v);
-    let ACCURACY = 0.25;
-    let Q_ACCURACY = ACCURACY * 0.1;
-    let REM_ACCURACY = ACCURACY - Q_ACCURACY;
-    let MAX_HYPOT2 = 432.0 * Q_ACCURACY * Q_ACCURACY;
-    let scaled_sqrt_tol = sqrt(REM_ACCURACY / scale);
-    // Fudge the subdivision count metric to account for `scale` when the subdivision is done in local
-    // coordinates.
-    var n_quads = max(u32(ceil(pow(err * (1.0 / MAX_HYPOT2), 1.0 / 6.0)) * scale), 1u);
-    n_quads = min(n_quads, MAX_QUADS);
-
-    var keep_params: array<SubdivResult, MAX_QUADS>;
-    var val = 0.0;
-    var qp0 = p0;
-    let step = 1.0 / f32(n_quads);
-    for (var i = 0u; i < n_quads; i += 1u) {
-        let t = f32(i + 1u) * step;
-        let qp2 = eval_cubic(p0, p1, p2, p3, t);
-        var qp1 = eval_cubic(p0, p1, p2, p3, t - 0.5 * step);
-        qp1 = 2.0 * qp1 - 0.5 * (qp0 + qp2);
-
-        // TODO: Estimate an accurate subdivision count for strokes
-        let params = estimate_subdiv(qp0, qp1, qp2, scaled_sqrt_tol);
-        keep_params[i] = params;
-        val += params.val;
-        qp0 = qp2;
+    // Drop zero length lines.
```

Comment:
Maybe mention here why the exact matches are valid (e.g. any invariants about prior detection of zero-length lines).